adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -3051,4 +3242,44 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
       loadParquetFragments(pathName, optPredicate = optPredicate, optProjection = optProjection)
     }
   }
+
+  /**
+   * Return integer size of partitions if the specified path of Parquet + Avro files is partitioned.
+   *
+   * @param pathName Path in which to look for partitioned flag.
+   * @return Return integer genomic base pair size of partitions
+   *
+   * If a glob is used, all directories within the blog must be partitioned, and must have been saved
+   * using the same partitioned bin size.  Behavior is undefined if this requirement is not met.
+   */
+  private def getPartitionedBinSize(pathName: String): Int = {
+
+    val partitionSizes = getFsAndFilesWithFilter(pathName, new FileFilter("_isPartitionedByStartPos")).map(f => {
+      val is = f.getFileSystem(sc.hadoopConfiguration).open(f)
+      val partitionSize = is.readInt
+      is.close()
```

Comment:
Might want to try with resources this IO block.  I hacked up [TryWith.scala](https://github.com/heuermh/biojava-adam/blob/master/src/main/scala/org/biojava/nbio/adam/TryWith.scala) in another repo, and was thinking of bringing it over for #1719.

Reply:
I'm fine with this code as is.

@heuermh perhaps you could move `TryWith` into bdg-utils and we could modify this as part of the push towards https://github.com/bigdatagenomics/adam/issues/1719 in 0.25.0?

Reply:
+1 @fnothaft